sdk/src/types/block/semantic.rs

```diff
@@ -149,7 +149,7 @@ impl TryFrom<u8> for TransactionFailureReason {
 ///
 pub struct ValidationContext<'a> {
     pub(crate) transaction: &'a Transaction,
-    pub(crate) transaction_hash: [u8; 32],
+    pub(crate) transaction_hash: TransactionHash,
```

Comment:
I am not fully sure what is being signed actually is the TransactionHash (anymore?). I asked here https://github.com/iotaledger/tips/pull/147/files#r1370380908.

Reply:
See the answer, as I suspected, this is not the same thing that is signed

Reply:
that's quite annoying. So we need two kinds of transaction hash?

Reply:
If you want it properly typed, yeah. At the very list a new method to get these bytes

Reply:
TransactionSigningHash? TransactionSignHash?

Reply:
I added a new type. I am not sure about all of the places that use the other hash so I added TODOs